src/ngraph/frontend/onnxifi/backend.cpp

```diff
@@ -0,0 +1,295 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+
+#include <onnxifi.h>
+
+#include "backend.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        void Backend::get_capabilities(void* infoValue, std::size_t* infoValueSize) const
+        {
+            if (infoValueSize == nullptr)
+            {
+                throw std::invalid_argument{"null pointer"};
+            }
+            std::size_t requested{*infoValueSize};
+            *infoValueSize = sizeof(::onnxEnum);
+            if (requested < *infoValueSize)
+            {
+                throw std::length_error{"not enough space"};
+            }
+            if (infoValue != nullptr)
+            {
+                *reinterpret_cast<::onnxEnum*>(infoValue) = ONNXIFI_CAPABILITY_THREAD_SAFE;
+            }
+        }
+
+        void Backend::get_device(void* infoValue, std::size_t* infoValueSize) const
+        {
+            const constexpr char name[] = "nGraph ";
```

Comment:
`const string` is the instantiation of an object (runtime). Here `constexpr` works better because all is resolved at compile time. `const constexpr` is needed for gcc 4.8 because it has initial support for `constexpr` which are actually not `const`. Until recently MSVC had the same limitation.

Reply:
performance is of no concern in this function. const char* would be fine with no real overhead.